pkg/util/instancemetadata/dev.go

```diff
@@ -10,15 +10,17 @@ import (
 	"github.com/Azure/go-autorest/autorest/azure"
 )
 
-func NewDev() (InstanceMetadata, error) {
-	for _, key := range []string{
-		"AZURE_SUBSCRIPTION_ID",
-		"AZURE_TENANT_ID",
-		"LOCATION",
-		"RESOURCEGROUP",
-	} {
-		if _, found := os.LookupEnv(key); !found {
-			return nil, fmt.Errorf("environment variable %q unset (development mode)", key)
+func NewDev(skipEnvCheck bool) (InstanceMetadata, error) {
+	if !skipEnvCheck {
```

Comment:
Rather than checking if a negatively named boolean is _not_ true, how about we name the parameter `envCheck`? That would change calls like 
```
env, err := env.NewCoreForCI(ctx, log, false)
                                         ^ "don't not check"
```
to 
```
env, err := env.NewCoreForCI(ctx, log, true)
                                         ^ "do check"
```